File: LoadBalancer.rb
require "socket"

loadbalancer=TCPServer.open("192.168.175.129",2000)
while true
client=loadbalancer.accept
rhost=client.peeraddr.last
puts "#message from client: #{fromclient}"
if fromclient=="fat"
sockettoserver1=TCPSocket.open("192.168.15.128",3000)
sockettoserver1.write(fromclient)
sockettoserver1.close

else if fromclient=="thin"
sockettoserver2=TCPSocket.open("192.168.15.129",4000)
sockettoserver2.write(fromclient)
sockettoserver2.close

else 
puts "ERROR"

end
end
end

File: LoadBalancer.java
import java.io.*;
import java.net.*;
public class LoadBalancer{
  public static void main(String []args)throws IOException{
  ServerSocket s1=new ServerSocket(1234);
  while(true){
    Socket connectionsocket=s1.accept();
    BufferedReader messagefromclient=new BufferedReader(new InputStreamReader(connectionsocket.getInputStream()));
    String clientsentence=messagefromclient.readLine();
    if(clientsentence.contains("Fat")){
    Socket server=new Socket("192.168.15.128,2345);
    DataOutputStream outtoserver=new DataOutputStream(server.getOutputStream());
    outtoserver2.writeBytes(clientsentence+"\n);
    server.close();
    }
   
    else if(clientsentence.contains("Thin")){
    Socket server=new Socket("192.168.15.129,2346);
    DataOutputStream outtoserver=new DataOutputStream(server.getOutputStream());
    outtoserver2.writeBytes(clientsentence+"\n);
    server.close();
    }
  
    else{System.out.println("ERROR"); }
    }
   }
}
